config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -52,16 +49,22 @@ public BundleMetadata build(final AnnotatedBundle annotatedBundle, final Bundle
             if (StringUtils.isNotBlank(projectInfo.getVersion())) {
                 name =  bundleName.substring(0, bundleName.indexOf(projectInfo.getVersion()) - 1);
             }
-            final String metadataId = StringUtils.isBlank(annotatedEntity.getMetadataId()) ? idGenerator.generate() :
-                    annotatedEntity.getMetadataId();
 
-            BundleMetadata.Builder builder = new BundleMetadata.Builder(encass.getType(), metadataId, name,
-                    projectInfo.getGroupName(), projectInfo.getVersion());
+            BundleMetadata.Builder builder = new BundleMetadata.Builder(encass.getType(), name, projectInfo.getName(), projectInfo.getGroupName(), projectInfo.getVersion());
             builder.description(annotatedEntity.getDescription());
-            builder.environmentEntities(getEnvironmentDependenciesMetadata(dependentEntities));
+            Collection<Metadata> referencedEntities = getEnvironmentDependenciesMetadata(dependentEntities);
+            builder.referencedEntities(referencedEntities);
+            if (!referencedEntities.isEmpty()) {
+                DependentBundle envBundle = new DependentBundle();
+                envBundle.setGroupName(projectInfo.getGroupName());
+                envBundle.setName(projectInfo.getName());
+                envBundle.setVersion(projectInfo.getVersion());
+                envBundle.setType("bundle");
+                annotatedBundle.getDependentBundles().add(envBundle);
+            }
```

Comment:
Why is this piece of code in MetadataBuilder? Can we move this initialization outside of this Builder as MetadataBuilder should only be building metadata and not initializing annotated bundle?

Reply:
done